Rewrite of test_lookup_query and slight refactoring of LookupQuery.

mtlynch opened the pull request:
This change refactors LookupQuery to make private the methods and
properties that are not referenced outside of the class itself (or the
test code). LookupQuery only needs to expose
initialize_from_http_request, so the rest of its methods are now
private.

This change includes a complete rewrite of test_lookup_query so that
it's only testing LookupQuery's public interface instead of its private
methods.

In the process of rewriting the test suite, I identified some bugs in
LookupQuery and also reduced some of the logging messages, because it
was firing warnings when optional fields were not set (the warnings
should only fire for non-optional fields or when an invalid option is
specified). I fixed a couple small bugs in this CL, but I left a few
TODOs in the test code for additional fixes I plan to make in a
subsequent CL.

mtlynch committed:
Rewrite of test_lookup_query and slight refactoring of LookupQuery.

This change refactors LookupQuery to make private the methods and
properties that are not referenced outside of the class itself (or the
test code). LookupQuery only needs to expose
initialize_from_http_request, so the rest of its methods are now
private.

This change includes a complete rewrite of test_lookup_query so that
it's only testing LookupQuery's public interface instead of its private
methods.

In the process of rewriting the test suite, I identified some bugs in
LookupQuery and also reduced some of the logging messages, because it
was firing warnings when optional fields were not set (the warnings
should only fire for non-optional fields or when an invalid option is
specified). I fixed a couple small bugs in this CL, but I left a few
TODOs in the test code for additional fixes I plan to make in a
subsequent CL.

yachang commented on server/mlabns/tests/test_lookup_query.py:
```diff
@@ -1,26 +1,68 @@
 
 from mlabns.util import constants
 from mlabns.util import lookup_query
+from mlabns.util import maxmind
```
Is this the geoip2.database from MaxMind?

mtlynch replied:
Yep

yachang commented on server/mlabns/tests/test_lookup_query.py:
```diff
@@ -29,562 +71,385 @@ def testDefaultConstructor(self):
-        query.set_ip_address_and_address_family(self.mock_request)
-        self.assertEqual('user_defined_ip', query.ip_address)
-        self.assertIsNone(query.address_family)
+    # TODO(mtlynch): These tests fail because LookupQuery actually doesn't
```
use """ to comment chunk of code.

mtlynch replied:
Fixed

yachang commented on server/mlabns/tests/test_lookup_query.py:
```diff
@@ -29,562 +71,385 @@ def testDefaultConstructor(self):
+        # defined field is the country name, the policy should implicitly be
+        # message.COUNTRY.
         self.assertEqual(message.POLICY_GEO, query.policy)
+        self.assertIsNone(query.latitude)
```
Add comments that why you set the value in line 282, but no value here.

mtlynch replied:
Fixed

mtlynch committed:
Making user_defined_country and user_defined_af public again.

I realized that outside classes depend on these two fields so they need
to be public for now, but I've added TODOs for plans to remove them
since they shoudn't be part of LookupQuery's public interface.

mtlynch committed:
Fixing test_lookup_query readability issues.

Fixing based on code review.

mtlynch commented:
Updated based on your notes.

I also realized that outside classes actually need access to `user_defined_af` and `user_defined_country` so I made them public again. Both of those fields are used really bizarrely and inconsistently so I added notes to fix them.

I made that fix in one commit and addressed your notes in another. You can look commit by commit or at a merged view of both commits:

https://github.com/mtlynch/mlab-ns-staging/compare/70429a122f852c32fabf4e391e5e89d780a0f862...e12d56241a9008f3216c0fa1fd95c0961a9a5b16

mtlynch committed:
Fixing one last reference to user_defined_af

yachang commented:
LGTM now.

mtlynch commented:
Thanks for the review!
